src/Approximations/template_directions.jl

```diff
@@ -330,3 +337,127 @@ end
 
 end # @eval
 end # if
+
+# ==================================================
+# Spherical directions
+# ==================================================
+
+@inline _isapprox_included(d, S) = any(isapproxzero.([norm(d - si, Inf) for si in S])) 
+
+"""
+    SphericalDirections <: AbstractDirections{Float64}
+
+Spherical directions representation.
+
+### Fields
+
+- `Nθ`    -- length of the partition of the acimutal angle
+- `Nφ`    -- length of the partition of the polar angle
+- `stack` -- list of computed directions
+
+### Notes
+
+The `SphericalDirections` constructor provides a sample of the unit sphere
+in ``\\mathbb{R}^3``, which is parameterized by the acimutal and polar angles
+``θ ∈ Dθ := [-π, π]`` and ``φ ∈ Dφ := [0, 2π]`` respectively. The domains ``Dθ``
+and ``Dφ`` are discretized in ``Nθ`` and ``Nφ`` respectively. Then the Cartesian
+componentes of each direction are obtained with
+
+```math
+[sin(θᵢ)*cos(φᵢ), sin(θᵢ)*sin(φᵢ), cos(θᵢ)]
+```
+Note that by discretization, the polar point may be computed more than once, in
+general. Removing duplicates takes time; you can choose to remove duplicates or
+not by setting the keyword argument `remove_duplicates` in the constructor
+(default, `true`).
+
+### Examples
+
+A `SphericalDirections` can be built in different ways. If you pass only one integer,
+it is used to discretize both ``θ`` and ``φ``:
+
+```jldoctest spherical_directions
+julia> using LazySets.Approximations: SphericalDirections
+
+julia> sd = SphericalDirections(3)
+SphericalDirections(3, 3, Array{Float64,1}[[-1.22465e-16, -0.0, -1.0], [0.0, 0.0, 1.0]])
+
+julia> sd.Nθ, sd.Nφ 
+(3, 3)
+```
+
+Repeated directions are removed by default. Use the keyword argument `remove_duplicates`
+to control this behavior:
+
+```jldoctest spherical_directions
+julia> sd_with_duplicates = SphericalDirections(3, remove_duplicates=true);
+
+julia> length(sd_with_duplicates)
+2
+
+julia> sd_without_duplicates = SphericalDirections(3, remove_duplicates=false);
+
+julia> length(sd_without_duplicates)
+9
+```
+
+Pass two integers to control the discretization in ``θ`` and in ``φ`` separately:
+```jldoctest spherical_directions
+julia> sd_4_5 = SphericalDirections(4, 5);
+
+julia> length(sd_4_5)
+5
+
+julia> sd_4_8 = SphericalDirections(4, 8);
+
+julia> length(sd_4_8)
+15
+```
+"""
+struct SphericalDirections <: AbstractDirections{Float64}
+    Nθ::Int
+    Nφ::Int
+    stack::Vector{Vector{Float64}} # stores the spherical directions
+
+    function SphericalDirections(Nθ::Int, Nφ::Int, stack::Vector{Vector{Float64}}; remove_duplicates=true)
```

Comment:
What is the reason to make `stack` an input argument? Should this not always be initialized to an empty vector?

Reply:
True, it doesn't need to be an input.